src/robusta/integrations/discord/sender.py

```diff
@@ -0,0 +1,281 @@
+from ...core.model.env_vars import DISCORD_TABLE_COLUMNS_LIMIT, ROBUSTA_LOGO_URL
+
+ACTION_TRIGGER_PLAYBOOK = "trigger_playbook"
+DiscordBlock = Dict[str, Any]
+SEVERITY_EMOJI_MAP = {
+    FindingSeverity.HIGH: ":red_circle:",
+    FindingSeverity.MEDIUM: ":orange_circle:",
+    FindingSeverity.LOW: ":yellow_circle:",
+    FindingSeverity.INFO: ":green_circle:",
+}
+SEVERITY_COLOR_MAP = {
+    FindingSeverity.HIGH: "14495556",
+    FindingSeverity.MEDIUM: "16027661",
+    FindingSeverity.LOW: "16632664",
+    FindingSeverity.INFO: "7909721",
+}
+MAX_BLOCK_CHARS = 2000
+MAX_FIELD_CHARS = 1024
+BLANK_CHAR = "\u200b"
+
+class DiscordDescriptionBlock(BaseBlock):
+    """
+    Discord description block
+    """
+    description: str
+
+
+class DiscordFieldBlock(BaseBlock):
+    """
+    Discord field block
+    """
+    name: str
+    value: str
+    inline: bool
+
+    def __init__(self, name: str, value: str, inline: bool = False):
+        value = DiscordSender.apply_length_limit(value)
+        super().__init__(name=name, value=value, inline=inline)
+
+
+def _add_color_to_block(block: DiscordBlock, msg_color: str):
+    return {**block, "color": msg_color}
+
+
+class DiscordSender:
+    def __init__(
+            self, url: str, cluster_name: str
+    ):
+        """
+        Set the Discord webhook url.
+        """
+        self.url = url
+        self.cluster_name = cluster_name
+
+    @classmethod
+    def __add_log_level_icon(cls, title: str, severity: FindingSeverity) -> str:
+        icon = SEVERITY_EMOJI_MAP.get(severity, "")
+        return f"{icon} {severity.name} - {title}"
+
+    @staticmethod
+    def apply_length_limit(msg: str, max_length: int = MAX_FIELD_CHARS):
+        if len(msg) <= max_length:
+            return msg
+        truncator = "..."
+        return msg[: max_length - len(truncator)] + truncator
+
+    @staticmethod
+    def __extract_markdown_name(block: MarkdownBlock):
+        title = BLANK_CHAR
+        text = block.text
+        regex = re.compile(r"\*.+\*")
+        match = re.match(regex, block.text)
+        if match:
+            title = text[match.span()[0]:match.span()[1]]
+            text = text[match.span()[1]:]
+        return title, DiscordSender.__transform_markdown_links(text) or BLANK_CHAR
+
+    @staticmethod
+    def __transform_markdown_links(text: str):
+        regex = re.compile(r"<.+>")
+        finds = re.findall(regex, text)
+        for find in finds:
+            cropped_url = find.replace("<", "").replace(">", "")
+            url, url_text = cropped_url.split("|")
+            new_url = f"[{url_text}]({url})"
+            text = text.replace(find, new_url)
+        return text
+
+    @staticmethod
+    def __format_final_message(discord_blocks: List[DiscordBlock],
+                               header_block: DiscordBlock, msg_color: Union[str, int]) -> Dict:
+        fields = [block for block in discord_blocks if 'value' in block]
```

Comment:
Is this check is to check if it's a DiscordFieldBlock?
If so, it would be better to use isinstance(block, DiscordFieldBlock)

Reply:
They are all converted into DiscordBlock in __to_discord method, so here isinstance will fail (as both DiscordField and Description blocks are inherited from BaseBlock and are not the instances of DiscordBlock) 
That's why I'm checking by key present in DiscordBlock - that is a dictionary basically

Reply:
I think this is a little bug prone, if someone will add another block with 'value' or 'description' field
Can we pass the original blocks into the `__format_final_message` function, and then, do the __to_dicsord inside?

It will allow us to check using isInstance
WDYT?

Reply:
Now it should be possible to check type of block using `isinstance` method